sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -2523,6 +2552,50 @@ val SHUFFLE_COMPRESSION_LZ4_CHUNK_SIZE = conf("spark.rapids.shuffle.compression.
       }
     }
   }
+
+  object Format extends Enumeration {
+    type Format = Value
+    val PLAIN, JSON = Value
+  }
+
+  def dumpConfigsWithDefault(args: Array[String]): Unit = {
```

Comment:
Nit: This should take separate arguments, have `main` deal with pulling them out to call this properly rather than hiding the call signature here.

Reply:
Yeah that would be better, but this logic cannot be in the `DumpDefaultConfigs.main()` as the `withResource` arm currently has a dependency on some spark stuff, which triggers shimming.

Reply:
I'm not asking for the _logic_ to be in main, I'm asking for main to pass the arguments separately.  Main just has to unpack the strings from the args[] array rather than having dumpConfigsWithDefault take an unclear and error-prone argument list.